rfc/rfc018-concent-credential.md

```diff
@@ -0,0 +1,155 @@
+
+### Status
+
+This document is currently a draft.
+
+### Copyright Notice
+
+![](../.gitbook/assets/license.png)
+
+This document is released under the [Attribution-ShareAlike 4.0 International \(CC BY-SA 4.0\) license](https://creativecommons.org/licenses/by-sa/4.0/).
+
+## 1. Introduction
+
+[RFC014](rfc014-authorization-credential.md) makes it possible for a custodian to create an authorization for an actor to access personal data. The problem with this is that it's not always known to the custodian when a authorization needs to be created. It also requires an interaction with the person the data belongs to, to authorize the data sharing. RFC014 also describes some of the cases in which the authorization credential alone does not support the case completely. One case in particular is of interest:
+
+* A patient at the GP's office wants the GP to view data from the hospital.
+
+The Nuts Consent Credential allows for the actor to be a witness to the patient's consent. The resulting credential can then be send to the custodian (via a Verifiable Presentation for example). The custodian can then decide to grant access to the actor automatically or use a custom workflow to decide on the result. The fact the actor acts as a witness also means it puts its reputation on the line.
+
+This RFC builds upon [RFC011](rfc011-verifiable-credential.md).
+
+## 2. Terminology
+
+* **DID**: [Decentralized Identifiers](https://www.w3.org/TR/did-core/).
+* **Policy**: An access policy defined by a Bolt. It describes the access to and operations on resources that are allowed. 
+* **VC**: [Verifiable Credential Data Model](https://www.w3.org/TR/vc-data-model/).
+* **VP**: Verifiable Presentation as described in the [Verifiable Credential Data Model](https://www.w3.org/TR/vc-data-model/).
+
+Other terminology comes from the [Nuts Start Architecture](rfc001-nuts-start-architecture.md#nuts-start-architecture).
+
+## 3. Credential data model
+
+### 3.1 Credential fields
+
+The `issuer` field of the credential MUST contain the DID of the actor. The `type` MUST include both the values `VerifiableCredential` and `NutsConsentCredential`.
+Bolts that use this credential MUST specify the maximum validity of the credential.
+
+### 3.2 CredentialSubject
+
+The `credentialSubject` field contains the following:
+
+```javascript
+{
+  "id": "did:nuts:SjkuVHVqZndMVVJwcnUzbjhuZklhODB1M1M0LW9LcWY0WUs5S2",
+  "custodian": "did:nuts:EgFjg8zqN6eN3oiKtSvmUucao4VF18m2Q9fftAeANTBd",
+  "purposeOfUse": "test-service",
+  "subject": "123456780"
+}
```

Comment:
Add a `expire` to the `credentialSubject`. It is is also in the signature, but that is imho about the signature and not about the actual consent.